LibraBFT/Impl/Consensus/RoundManager.agda

```diff
@@ -95,6 +98,106 @@ addVoteM now vote = do
       _ →
         pure unit
 
+  processVote : Instant → VoteMsg → LBFT Unit
+  processVote now msg = pure unit
+
+  ------------------------------------------------------------------------------
+  syncUpM : Instant → SyncInfo → Author → LBFT (ErrLog ⊎ Unit)
+  ensureRoundAndSyncUpM : Instant → Round → SyncInfo → Author → Bool →
+                          LBFT (ErrLog ⊎ Bool)
+  processProposalM : Block → LBFT Unit
+  executeAndVoteM : Block → LBFT (ErrLog ⊎ VoteWithMeta)
+
+  -- external entry point
+  -- TODO-2: The sync info that the peer requests if it discovers that its round
+  -- state is behind the sender's should be sent as an additional argument, for now.
+
+  -- This is broken up into smaller pieces in order to aid in the verification effort.
+  processProposalMsgM-check₁ : Instant → ProposalMsg → Author → LBFT Unit
+  processProposalMsgM-check₁-cont : Instant → ProposalMsg → Author → ErrLog ⊎ Bool → LBFT Unit
+
+  processProposalMsgM : Instant → {- Author → -} ProposalMsg → LBFT Unit
+  processProposalMsgM now {- from -} pm
+     with pm ^∙ pmProposer
+  ...| nothing = pure unit -- errorExit "ProposalMsg does not have an author"
+  ...| just auth =
+    processProposalMsgM-check₁ now pm auth
+
+  processProposalMsgM-check₁ now pm auth =
+    ensureRoundAndSyncUpM now (pm ^∙ pmProposal ∙ bRound) (pm ^∙ pmSyncInfo) auth true
+      >>= processProposalMsgM-check₁-cont now pm auth
+
+  processProposalMsgM-check₁-cont now pm auth = λ where
+    (inj₁ _)    → pure unit -- log: error: <propagate error>
+    (inj₂ true) → processProposalM (pm ^∙ pmProposal)
+    (inj₂ false) → pure unit -- log: info: dropping proposal for old round
+
+  syncUpM now syncInfo author = ok unit
+
+-- ensureRoundAndSyncUp
+-----------------------
+
+  ensureRoundAndSyncUpM-check₁ : Instant → Round → SyncInfo → Author → Bool →
+                                 LBFT (ErrLog ⊎ Bool)
+  ensureRoundAndSyncUpM-check₁-cont : Round → Unit → LBFT (ErrLog ⊎ Bool)
+
+  ensureRoundAndSyncUpM now messageRound syncInfo author helpRemote = do
+    currentRound ← use (lRoundState ∙ rsCurrentRound)
+    if ⌊ messageRound <? currentRound ⌋
+      then ok false
+      else ensureRoundAndSyncUpM-check₁ now messageRound syncInfo author helpRemote
+
+  ensureRoundAndSyncUpM-check₁ now messageRound syncInfo author helpRemote = do
+    syncUpM now syncInfo author ∙?∙ ensureRoundAndSyncUpM-check₁-cont messageRound
+
+  ensureRoundAndSyncUpM-check₁-cont messageRound = λ _ → do
+    currentRound' ← use (lRoundState ∙ rsCurrentRound)
+    if not ⌊ messageRound ≟ℕ currentRound' ⌋
+      then bail unit  -- error: after sync, round does not match local
+      else ok true
+
+  processProposalM proposal = do
+    _rm ← get
```

Comment:
This was an attempt to call attention to the fact that `_rm` is not bound in the Haskell implementation. I will add a comment here explaining the issue, but is there a naming convention we could use for these situations (which arise because we do not have dependent lenses, so some fields of `RoundManager` must be set this way)?

Reply:
No convention (yet?).
I would just do the `s <- get` or `gets` would be even better.